lending_pool/src/lib.rs

```diff
@@ -75,16 +72,62 @@ pub trait LendingPool:
         Ok(())
     }
 
+    #[payable("*")]
+    #[endpoint(borrow)]
+    fn borrow_endpoint(
+        &self,
+        #[payment_token] payment_lend_id: TokenIdentifier,
+        #[payment_nonce] payment_nonce: u64,
+        #[payment_amount] payment_amount: Self::BigUint,
+        collateral_token_id: TokenIdentifier,
+        asset_to_borrow: TokenIdentifier,
+        #[var_args] caller: OptionalArg<Address>,
+    ) -> SCResult<()> {
+        let initial_caller = self.caller_from_option_or_sender(caller);
+
+        self.require_amount_greater_than_zero(&payment_amount)?;
+        self.require_non_zero_address(&initial_caller)?;
+
+        let borrow_token_pool_address = self.get_pool_address_non_zero(&asset_to_borrow)?;
+        let lend_token_pool_address = self.get_pool_address_non_zero(&payment_lend_id)?;
+
+        let metadata = self
+            .liquidity_pool_proxy(lend_token_pool_address.clone())
+            .get_interest_metadata(payment_nonce)
+            .execute_on_dest_context();
+
+        let ltv = self.asset_ltv(&collateral_token_id).get();
```

Comment:
Maybe check if this is zero and return an error if so. That way, we don't even make the call if the mapping doesn't exist. (which would yield wrong results anyway)

Reply:
done. added `get_ltv_exists_and_non_zero`